core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -146,4 +148,47 @@ public void testNotPresentFolder() throws Exception {
     workflowManager.waitForFinish(5, TimeUnit.MINUTES);
     Assert.assertEquals("FAILED", workflowManager.getHistory().get(0).getStatus().name());
   }
+
+  @Test
+  public void testRecursiveFolders() throws Exception {
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put(Constants.Reference.REFERENCE_NAME, "TestCase")
+      .put(Properties.File.PATH, "src/test/resources/")
+      .put(Properties.File.FILE_REGEX, "fileBatchSource.*")
+      .put(Properties.File.IGNORE_NON_EXISTING_FOLDERS, "false")
+      .put(Properties.File.FILESYSTEM_PROPERTIES,
+           "{\"mapreduce.input.fileinputformat.input.dir.recursive\": \"true\" }")
```

Comment:
We should have a Properties.File.RECURSIVE property here, because we don't want expose Hadoop properties here. Recursive read is common enough to justify our own property.